src/CoreWCF.Primitives/src/CoreWCF/Description/TypeLoader.cs

```diff
@@ -221,7 +221,10 @@ private void AddAuthorizeOperations(ContractDescription contractDesc, bool implI
                         GetIOperationAttributesFromType<IAuthorizeOperation>(opDesc, targetIface, currentType);
                         for (int j = 0; j < toAdd.Count; j++)
                         {
-                            opDesc.AuthorizeOperation.Add(toAdd[j]);
+                            if (!opDesc.AuthorizeOperation.Contains(toAdd[j].GetType()))
```

Comment:
Not adding it can at best cause confusion, and at worst could cause a security problem for someone. If they are expecting authorization to work one way and are unaware that a base type already did this, this could result in different authorization behavior happening than expected. We should fail and throw an exception in this scenario.

Reply:
I worked out what the problem was. It recursively gets each of the inherited contract descriptions. What was happening is it creates the derived contract descriptions first (as they need to be added to the top level one so need to be created first). When it creates the contract description for the inherited contract interface, it iterates over every operation and looks for that method on the implementation class. Once it has that method, it looks for the attributes and populates the AuthorizeOperation on the OperationDescription. So far so good. Then when it has finished constructing all the inherited contract descriptions, when it's creating the main ContractDescription, it copies each of the OperationDescriptions from the inherited contracts into the main one. So now the main ContractDescription has all the inherited OperationDescriptions in its list of operations. It then goes to populate the authorization info for the main contract, and duplicates the work. The change that needs to be made is to check if the operations declared contract description is the same as the currently being processed contract description, and if they are different, skip it. Below is the correct code.  
```c#
        private void AddAuthorizeOperations(ContractDescription contractDesc, bool implIsCallback, ContractReflectionInfo reflectionInfo)
        {

            for(int i=0; i< contractDesc.Operations.Count; i++)
            {
                OperationDescription opDesc = contractDesc.Operations[i];
                if (opDesc.DeclaringContract != contractDesc)
                {
                    continue;
                }

                Type targetIface = implIsCallback ? opDesc.DeclaringContract.CallbackContractType : opDesc.DeclaringContract.ContractType;
                ApplyServiceInheritance(
                    opDesc.AuthorizeOperation,
                    delegate (Type currentType, KeyedByTypeCollection<IAuthorizeOperation> behaviors)
                    {
                        KeyedByTypeCollection<IAuthorizeOperation> toAdd =
                        GetIOperationAttributesFromType<IAuthorizeOperation>(opDesc, targetIface, currentType);
                        for (int j = 0; j < toAdd.Count; j++)
                        {
                            opDesc.AuthorizeOperation.Add(toAdd[j]);
                        }
                    });
            }
        }
```

The tests around this are have been disabled from our CI runs. I want to do the work to enable them, and have a test which represents your scenario to ensure this doesn't ever regress. I'm going to close this PR as I'll create one with this fix and all the tests for it. I posted the code here in case you are running with your own fork so you can use the correct fix.  

Your fix would have worked and would have matched the behavior we have for any other attributes if it wasn't for this being a security feature where ignoring an attribute can introduce security problems. Ironically there's a "bug" in this code which stopped this from resulting in an urgent security release. It directly adds the operations to `opDesc.AuthorizeOperations` in the delegate. Everywhere else using ApplyServiceInheritance should be adding things to the `behaviors` collection that's passed as the second parameter to the delegate, and that silently drops any duplicates, just like your proposal did. Because this code "incorrectly" added directly to the collection, it failed if there were two conflicting attributes in the implementation type hierarchy, which in this case, and only this case, we actually want.